File: eform-client/src/app/common/services/advanced/workers.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import {
  OperationDataResult,
  OperationResult,
  WorkerCreateModel,
  WorkerDto,
  WorkerModel,
} from 'src/app/common/models';
import { ApiBaseService } from 'src/app/common/services';

const WorkersMethods = {
  GetAll: '/api/workers/index',
  GetSingle: '/api/workers/edit',
  UpdateSingle: '/api/workers/update',
  CreateSingle: '/api/workers/create',
  DeleteSingle: '/api/workers/delete',
};

@Injectable()
export class WorkersService {
  private apiBaseService = inject(ApiBaseService);


  getAllWorkers(): Observable<OperationDataResult<Array<WorkerDto>>> {
    return this.apiBaseService.get<Array<WorkerDto>>(WorkersMethods.GetAll);
  }

  getSingleWorker(id: number): Observable<OperationDataResult<WorkerDto>> {
    return this.apiBaseService.get<WorkerDto>(
      WorkersMethods.GetSingle + '/' + id
    );
  }

  updateSingleWorker(model: WorkerModel): Observable<OperationResult> {
    return this.apiBaseService.post<WorkerModel>(
      WorkersMethods.UpdateSingle,
      model
    );
  }

  deleteSingleWorker(id: number): Observable<OperationResult> {
    return this.apiBaseService.get(WorkersMethods.DeleteSingle + '/' + id);
  }

  createWorker(model: WorkerCreateModel): Observable<OperationResult> {
    return this.apiBaseService.post<WorkerCreateModel>(
      WorkersMethods.CreateSingle,
      model
    );
  }
}

File: eFromAPI/eFormAPI/eFormAPI.Core/Services/WorkersService.cs
using System;
using System.Collections.Generic;
using eFormAPI.Common.Infrastructure;
using eFormAPI.Common.Infrastructure.Helpers;
using eFormAPI.Common.Infrastructure.Models.API;
using eFormAPI.Common.Models;
using eFormAPI.Core.Abstractions;
using eFormShared;
using Microsoft.AspNetCore.Mvc;

namespace eFormAPI.Core.Services
{
    public class WorkersService : IWorkersService
    {
        private readonly IEFormCoreService _coreHelper;

        public WorkersService(IEFormCoreService coreHelper)
        {
            _coreHelper = coreHelper;
        }

        public OperationDataResult<List<Worker_Dto>> Index()
        {
            var core = _coreHelper.GetCore();
            var workersDto = core.Advanced_WorkerReadAll("not_removed", null, null);

            return new OperationDataResult<List<Worker_Dto>>(true, workersDto);
        }

        public OperationDataResult<Worker_Dto> Edit(int id)
        {
            var core = _coreHelper.GetCore();
            var workerDto = core.Advanced_WorkerRead(id);

            return new OperationDataResult<Worker_Dto>(true, workerDto);
        }

        public OperationResult Update(WorkerModel workerModel)
        {
            try
            {
                var core = _coreHelper.GetCore();
                var workerDto = core.Advanced_WorkerRead(workerModel.Id);
                var isUpdated = core.Advanced_WorkerUpdate(workerModel.Id, workerModel.UserFirstName,
                    workerModel.UserLastName, workerDto.Email);

                return isUpdated
                    ? new OperationResult(true, LocaleHelper.GetString("WorkerParamWasUpdated", workerModel.Id))
                    : new OperationResult(false, LocaleHelper.GetString("WorkerParamCantBeUpdated", workerModel.Id));
            }
            catch (Exception)
            {
                return new OperationResult(false, LocaleHelper.GetString("WorkerParamCantBeUpdated", workerModel.Id));
            }
        }


        public OperationResult Сreate(WorkerCreateModel model)
        {
            try
            {
                var core = _coreHelper.GetCore();
                var workerDto = core.Advanced_WorkerCreate(model.FirstName, model.LastName,
                    model.SiteId + "." + model.CustomerNo + "@invalid.invalid");
                var createdWorker =
                    core.Advanced_SiteWorkerCreate(new SiteName_Dto(model.SiteId, "", null, null), workerDto);

                return new OperationResult(true, LocaleHelper.GetString("WorkerWasSuccessfullyCreated"));
            }
            catch (Exception)
            {
                return new OperationResult(false, LocaleHelper.GetString("ErrorWhileCreatingWorker"));
            }
        }

        [HttpGet]
        public OperationResult Delete(int id)
        {
            try
            {
                var core = _coreHelper.GetCore();
                var workerDto = core.Advanced_WorkerRead(id);

                if (workerDto.Equals(null))
                {
                    return new OperationDataResult<SiteNameModel>(false,
                        LocaleHelper.GetString("SiteWithIdCouldNotBeDeleted", id));
                }

                return core.Advanced_WorkerDelete(id)
                    ? new OperationDataResult<SiteNameModel>(true,
                        LocaleHelper.GetString("WorkerParamDeletedSuccessfully", workerDto.FirstName,
                            workerDto.LastName))
                    : new OperationDataResult<SiteNameModel>(false,
                        LocaleHelper.GetString("WorkerParamCantBeDeted", workerDto.FirstName, workerDto.LastName));
            }

            catch (Exception)
            {
                return new OperationDataResult<SiteNameModel>(false,
                    LocaleHelper.GetString("SiteWithIdCouldNotBeDeleted", id));
            }
        }
    }
}